ably-ios/ARTRealtimePresence.m

```diff
@@ -48,33 +72,45 @@ - (void)enterClient:(NSString *)clientId data:(id)data cb:(ARTStatusCallback)cb
     msg.clientId = clientId;
     msg.data = data;
 
-    msg.connectionId = [self channel].realtime.connectionId;
+    msg.connectionId = [self channel].realtime.connection.id;
     [[self channel] publishPresence:msg cb:cb];
 }
 
-- (void)update:(id)data cb:(ARTStatusCallback)cb {
+- (void)update:(id)data {
+    [self update:data cb:nil];
+}
+
+- (void)update:(id)data cb:(void (^)(ARTErrorInfo * _Nullable))cb {
     [self updateClient:[self channel].clientId data:data cb:cb];
 }
 
-- (void)updateClient:(NSString *)clientId data:(id)data cb:(ARTStatusCallback)cb {
+- (void)updateClient:(NSString *)clientId data:(id)data {
+    [self updateClient:clientId data:data cb:nil];
+}
+
+- (void)updateClient:(NSString *)clientId data:(id)data cb:(void (^)(ARTErrorInfo * _Nullable))cb {
     ARTPresenceMessage *msg = [[ARTPresenceMessage alloc] init];
     msg.action = ARTPresenceUpdate;
     msg.clientId = clientId;
-    if(!msg.clientId) {
```

Comment:
Actually, you were right here. This method is not only public, it's also called from `update`, which passes the channels' `clientId` which _can_ be nil. Joke's on me for fiddling with internals without need... Fixed at 500d0f874e86759c7c6fa6c0f37821497b8b6e0d.